### ISSUE TITLE
Custom shop page slug and custom permalink structure not working

### ISSUE THREAD

**Issue by aaronslotnik** [woocommerce/woocommerce#15237]
### EXPLANATION OF THE ISSUE

Hello. Using WC 3.0.7, I've changed the shop page Title and Slug from 'Shop' to 'Marketplace'.  That has worked fine with the default permalink structure (/product/sample-product/). The shop page displays correctly and single product pages display correctly when selected (clicked) from the shop page.

However, when the product permalinks are changed to either 'Shop base' or 'Shop base with category', the single product pages do not display at all.  The permalink looks correct and conforms to the permalink structure setting, but when trying to navigate to that URL directly or from the shop page the shop page is redisplayed even though the URL is of the single product.

If I revert back to the delivered shop page Title and Slug from 'Marketplace' to 'Shop', then the custom permalinks work correctly.  This appears to be a bug and is different from other similar issues where a 404 error page is displayed.

I would appreciate for help and support as we would like to use a different shop page Title and Slug.  Thanks!

### STEPS TO REPRODUCE THE ISSUE

See above.

### SYSTEM STATUS

<details>
```
`
### WordPress Environment ###

Home URL: http://old-fashioned-cheetah.w4.poopy.life
Site URL: http://old-fashioned-cheetah.w4.poopy.life
WC Version: 3.0.7
Log Directory Writable: 
WP Version: 4.7.5
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: 
Language: en_US

### Server Environment ###

Server Info: Apache/2.4.6 ()
PHP Version: 5.6.30
PHP Post Max Size: 256 MB
PHP Time Limit: 600
PHP Max Input Vars: 1000
cURL Version: 7.36.0
OpenSSL/1.0.1e

SUHOSIN Installed: –
Max Upload Size: 256 MB
Default Timezone is UTC: 
fsockopen/cURL: 
SoapClient:  Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: 
GZip: 
Multibyte String: 
Remote Post: 
Remote Get: 

### Database ###

WC Database Version: 3.0.7
WC Database Prefix: wp_
woocommerce_sessions: 
woocommerce_api_keys: 
woocommerce_attribute_taxonomies: 
woocommerce_downloadable_product_permissions: 
woocommerce_order_items: 
woocommerce_order_itemmeta: 
woocommerce_tax_rates: 
woocommerce_tax_rate_locations: 
woocommerce_shipping_zones: 
woocommerce_shipping_zone_locations: 
woocommerce_shipping_zone_methods: 
woocommerce_payment_tokens: 
woocommerce_payment_tokenmeta: 
MaxMind GeoIP Database:  The MaxMind GeoIP Database does not exist - Geolocation will not function. You can download and install it manually from http://dev.maxmind.com/geoip/legacy/geolite/ to the path: . Scroll down to "Downloads" and download the "Binary / gzip" file next to "GeoLite Country". Please remember to uncompress GeoIP.dat.gz and upload the GeoIP.dat file only.

### Security ###

Secure connection (HTTPS): Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: 

### Active Plugins (12) ###

Beaver Builder Plugin (Standard Version): by The Beaver Builder Team – 1.10.4
EWWW Image Optimizer: by Shane Bishop – 3.3.1
Google Analytics for WordPress by MonsterInsights: by MonsterInsights – 6.1.10
Sandbox: by Soflyy – 1.0.0-beta-1.0
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com
DavidAnderson – 1.13.1

User Switching: by John Blackbourn – 1.0.9
WC Vendors: by WC Vendors – 1.9.11
WooCommerce Stripe Gateway: by WooCommerce – 3.1.8
WooCommerce: by Automattic – 3.0.7
Wordfence Security: by Wordfence – 6.3.9
Yoast SEO: by Team Yoast – 4.7.1
WP Super Cache: by Automattic – 1.4.9

### Settings ###

API Enabled: 
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #62 - /marketplace/
Cart: #63 - /cart/
Checkout: #64 - /checkout/
My account: #65 - /my-account/

### Theme ###

Name: GeneratePress Child
Version: 0.1
Author URL: http://edge22.com
Child Theme: 
Parent Theme Name: GeneratePress
Parent Theme Version: 1.3.46
Parent Theme Author URL: https://tomusborne.com
WooCommerce Support: 

### Templates ###

Overrides: –
`
```
</details>

**Comment by mikejolley** [comment]
This is a conflict. Give the categories a different base, ensure 'marketplace' is not an attribute (or give those a unique base too) and ensure the vendors plugin does not use 'marketplace' for anything itself.

 I have no such issue with that string locally: https://dl.dropboxusercontent.com/s/vm0utzxp3svy0n3/2017-05-23%20at%2020.14.png

**Comment by aaronslotnik** [comment]
Hello Mike - Thanks for your quick response.  Can you elaborate on what is a conflict?  I've attached a screenshot of my permalink settings--as you'll see I left the delivered Product category base as 'product-category'.  Did you change the delivered Shop page title and permalink? Could you provide a screenshot of that as well?  Although I see /marketplace in the URL for the single product, I still see Shop in the breadcrumb.  Thanks!

![wc_permalink_issue](https://cloud.githubusercontent.com/assets/28902171/26376991/825eec96-3fdc-11e7-93fc-051377e8a839.PNG)

**Comment by mikejolley** [comment]
just out of interest, try marketplace2 for your shop page.

**Comment by aaronslotnik** [comment]
I'm stumped. I changed the Shop page to Marketplace2 and that worked correctly. I changed it back to Marketplace and the issue reappeared. I deactivated the WC Vendors plugin, but the issue remains. What else could be causing the conflict? Thanks!

**Comment by mikejolley** [comment]
htaccess rules? Attributes like I said earlier? Add bases to all the categories and attributes on that screen ^

**Comment by aaronslotnik** [comment]
I didn't update htaccess, but have attached it. Added bases to all the categories and attributes on the Permalinks page . . . screenshot attached. Sadly, the issue remains . . . any other ideas? Thanks!
[htaccess_Backup_for_old_fashioned_cheetah.w4.poopy.life.txt](https://github.com/woocommerce/woocommerce/files/1023950/htaccess_Backup_for_old_fashioned_cheetah.w4.poopy.life.txt)
![wc_permalink_issue6](https://cloud.githubusercontent.com/assets/28902171/26380270/98b36c06-3feb-11e7-9e1f-e0ad172240f6.PNG)

**Comment by mikejolley** [comment]
So when you change to marketplace2, what does /marketplace/ show?

**Comment by aaronslotnik** [comment]
Good idea. The first time it took a little while and I got a 404 page but didn't get a screenshot.  After that, somehow it was automatically redirecting to /marketplace2/!  Then I changed the Shop page to 'Marketplaces' and the same thing happened, although I didn't even get a 404 the first time. I thought maybe Yoast was doing it so I deactivated that and it was still happening.  Then I changed the Shop page to 'Random' and got the 404 again . . . screenshot below.
![wc_permalink_issue7](https://cloud.githubusercontent.com/assets/28902171/26384248/e467d03c-4005-11e7-9963-bfe16e2fa7b3.PNG)

**Comment by mikejolley** [comment]
Odd. I'm not sure what to suggest at this point. I'd usually expect plugin/htaccess/cache. It looks like that endpoint must be used for something else.

**Comment by Garconis** [comment]
Having a similar problem, where adding this causes all page except the homepage to 404: https://freshysites.d.pr/uiygms

For now I decided to just use /item/ for the Custom base.

**Comment by okegima** [comment]
i did try with /item/ but when am going on, am having a 404 error... i did creat a page-item.php and in page a item. it work, but when am doing /item/page/2/ am having a 404 error... :(

**Comment by tassawer-hussain** [comment]
The issue exist in one of the following.

1. .htaccess file
2. Plugin Conflict
3. Same base use somewhere else or there is a page/post/CPT with same slug.

I was facing the same issue. I had changed the title and URL of the "Shop" page to "Services" as I am offering services instead of physical products.

The shop page is working fine and I have set the custom base for products i.e '/service/'

So my shop page URL is [Shop Page](https://2bytecode.com/services/)
and my product URL like [Product Page](https://2bytecode.com/service/wordpress-site-maintenance/)
